extensions/mssql/src/sparkFeature/dialog/sparkJobSubmission/sparkJobSubmissionService.ts

```diff
@@ -0,0 +1,187 @@
+                }
+            };
+
+            const response = await this._requestPromise(options);
+            if (response && response.log) {
+                return this.extractYarnAppIdFromLog(response.log);
+            }
+
+            return Promise.reject(localize('sparkJobSubmission_LivyNoLogReturned',
+            'No log is returned within response.{0}[Error] {1}', os.EOL, JSON.stringify(response)));
+        } catch (error) {
+            return Promise.reject(error);
+        }
+    }
+
+
+    private extractYarnAppIdFromLog(log: any): LivyLogResponse {
+        let logForPrint = log;
+        if (Array.isArray(log)) {
+            logForPrint = log.join(os.EOL);
+        }
+
+        // eg: '18/08/23 11:02:50 INFO yarn.Client: Application report for application_1532646201938_0182 (state: ACCEPTED)'
+        for (let entry of log) {
+            if (entry.indexOf('Application report for') >= 0 && entry.indexOf('(state: ACCEPTED)') >= 0) {
+                let tokens = entry.split(' ');
+                for (let token of tokens) {
+                    if (token.startsWith('application_')) {
+                        return new LivyLogResponse(logForPrint, token);
+                    }
+                }
+            }
+        }
+
+        return new LivyLogResponse(logForPrint, '');
+    }
+}
+
+export class SparkJobSubmissionInput {
+    public setSparkClusterInfo(hadoopConnection: SqlClusterConnection): void {
+        this._host = hadoopConnection.host;
+        this._port = hadoopConnection.port;
+        this._livyPath = constants.livySubmitPath;
+        this._userName = hadoopConnection.user;
+        this._passWord = hadoopConnection.pass;
+    }
+
+    constructor(
+        private readonly _jobName: string,
+        private readonly _sparkFile: string,
+        private readonly _mainClass: string,
+        private readonly _arguments: string,
+        private readonly _jarFileList: string,
+        private readonly _pyFileList: string,
+        private readonly _otherFileList: string,
+        private _host?: string,
+        private _port?: string,
+        private _livyPath?: string,
+        private _userName?: string,
+        private _passWord?: string) {
+    }
+
+    public get jobName(): string { return this._jobName; }
+    public get sparkFile(): string { return this._sparkFile; }
+    public get mainClass(): string { return this._mainClass; }
+    public get jobArguments(): string { return this._arguments; }
+    public get jarFileList(): string { return this._jarFileList; }
+    public get otherFileList(): string { return this._otherFileList; }
+    public get pyFileList(): string { return this._pyFileList; }
+    public get host(): string { return this._host; }
+    public get port(): string { return this._port; }
+    public get livyPath(): string { return this._livyPath; }
+    public get userName(): string { return this._userName; }
```

Comment:
username is generally all lowercase.

Reply:
fixed. changed to `user` since the term `user` is being used in spark request field.